devnet/run-polkadot.sh

```diff
@@ -0,0 +1,358 @@
+    --validator \
+    --tmp \
+    --rpc-cors all \
+    --unsafe-ws-external \
+    --unsafe-rpc-external \
+    --chain $dir/specs/polkadot-local.raw.json > $dir/logs/relay.charlie.out 2>&1 &
+  nohup $dir/bin/polkadot \
+    --dave \
+    --validator \
+    --tmp \
+    --chain $dir/specs/polkadot-local.raw.json > $dir/logs/relay.dave.out 2>&1 &
+  nohup $dir/bin/polkadot \
+    --eve \
+    --validator \
+    --tmp \
+    --chain $dir/specs/polkadot-local.raw.json > $dir/logs/relay.eve.out 2>&1 &
+
+
+
+#  grep -oP '(?<=seed:       0x)[^\n]+' specs/circuitb2.key
+  circuita1_seed=$(grep -oP '(?<=seed:       0x)[^\n]+' $dir/specs/circuita1.key)
+  circuita2_seed=$(grep -oP '(?<=seed:       0x)[^\n]+' $dir/specs/circuita2.key)
+  circuitb1_seed=$(grep -oP '(?<=seed:       0x)[^\n]+' $dir/specs/circuitb1.key)
+  circuitb2_seed=$(grep -oP '(?<=seed:       0x)[^\n]+' $dir/specs/circuitb2.key)
+
+  nohup $dir/bin/devnet-circuit-collator \
+    --port 33333 \
+    --ws-port 1933 \
+    --rpc-port 1833 \
+    --collator \
+    --base-path $dir/data/circuita1 \
+    --node-key "$circuita1_seed" \
+    --rpc-cors all \
+    --unsafe-ws-external \
+    --unsafe-rpc-external \
+    --chain $dir/specs/circuita.raw.json \
+    --execution Wasm \
+    -- \
+    --chain $dir/specs/polkadot-local.raw.json \
+    --discover-local > $dir/logs/circuit.a.out 2>&1 &
+  nohup $dir/bin/devnet-circuit-collator \
+    --port 33332 \
+    --ws-port 1932 \
+    --rpc-port 1832 \
+    --collator \
+    --base-path $dir/data/circuita2 \
+    --node-key "$circuita2_seed" \
+    --chain $dir/specs/circuita.raw.json \
+    --execution Wasm \
+    -- \
+    --chain $dir/specs/polkadot-local.raw.json \
+    --discover-local > $dir/logs/circuit.b.out 2>&1 &
+  nohup $dir/bin/devnet-circuit-collator \
+    --port 23333 \
+    --ws-port 2933 \
+    --rpc-port 2833 \
+    --collator \
+    --base-path $dir/data/circuitb1 \
+    --node-key "$circuitb1_seed" \
+    --chain $dir/specs/circuitb.raw.json \
+    --execution Wasm \
+    -- \
+    --chain $dir/specs/polkadot-local.raw.json \
+    --discover-local > $dir/logs/circuit.c.out 2>&1 &
+  nohup $dir/bin/devnet-circuit-collator \
+    --port 23332 \
+    --ws-port 2932 \
+    --rpc-port 2832 \
+    --collator \
+    --base-path $dir/data/circuitb2 \
+    --node-key "$circuitb2_seed" \
+    --chain $dir/specs/circuitb.raw.json \
+    --execution Wasm \
+    -- \
+    --chain $dir/specs/polkadot-local.raw.json \
+    --discover-local > $dir/logs/circuit.d.out 2>&1 &
+}
+
+devnet() {
+  root_dir=$(git rev-parse --show-toplevel)
+  mkdir -p $root_dir/devnet/{bin,data,specs}
+  mkdir -p $root_dir/devnet/data/{circuita1,circuita2,circuitb1,circuitb2}
+  build_nodes
+  keygen
+  build_relay_chain_spec
+  build_para_chain_specs
+  build_para_genesis_states
+  build_para_wasm_runtimes
+  rm -rf $dir/data/*
+  start_nodes
+  npx --yes wait-port -t 33000 localhost:1933
+  npx --yes wait-port -t 33000 localhost:1944
+#  set_keys
```

Comment:
without keys set the collators won't produce blocks
```suggestion
set_keys
```